### PR TITLE
Fix Idaho income tax brackets

### LINKED ISSUE #3588: Fix Idaho income tax brackets
In the 0.615.0 version, the ID income tax brackets are incorrect for 2021 and also (it would seem) for 2023.  These problems would have easily been detected in the development phase if sensible test-driven development practices were being practiced.   Add unit tests for 2021 and 2023 to adequately test the accuracy of the income tax bracket parameters.

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #3588

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:1]: could you update line 10 to indicate it includes 2023, and also that the legal code indicates HOH is treated as married

**Review by MaxGhenis — APPROVED** [review]
good to merge after this documentation suggestion (apply to other files too)

File: policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:1]: I can not find the legal code reference but we have been using the following section of the [instructions](https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-06-2023.pdf#page=9):
image

This indicates the tax threshold which is the same for HoH/Widow/Married

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/id/tax/income/main/head_of_household.yaml:1]: It's in 63-3024(2)(b)
>For the purposes of this section, a return of a surviving spouse, as defined in section 2(a) of the Internal Revenue Code, and a head of household, as defined in section 2(b) of the Internal Revenue Code, shall be treated as a joint return.

**Pull request merged**